{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4222c806-acee-4ec3-8fd9-1bc4f2d4c758",
   "metadata": {},
   "source": [
    "# **Alignment**\n",
    "\n",
    "In this example notebook, I will be exploring, explaining, and providing an example of the alignment of RNA-Seq data.\n",
    "\n",
    "RNA-Seq is a foundational methodology in Bioinformatics the careful consideration of data, QC, trimming, alignment and more, inferences can be made from RNA.\n",
    "\n",
    "RNA-seq tools are not standardized although there are highly creditable tools available. Due to the need for large amount of data, I will either find a small sample, or find a manner to include a fily I generate on my own to use as dummy data for this example.\n",
    "\n",
    "- [**Pre-requisites**](#Pre-requisites) -> Outlines what is required to follow this example.\n",
    "\n",
    "- [**Files**](#Files) -> Outlines the file types involved in alignment.\n",
    "\n",
    "- [**Data**](#Data) -> Shows where & how to get data (FASTQ)\n",
    "\n",
    "- [**QC**](#QC) -> Quality Check of the reads data (FastQC / MultiQC)\n",
    "\n",
    "- [**Trimming**](#Trimming) -> Optional step, explains how to trim reads\n",
    "\n",
    "- [**Alignment**](#Alignment) -> Alignment using STAR produces BAM file\n",
    "\n",
    "- [**Down Stream Analysis**](#Down-Stream-Analysis) -> Analyze data with various stats, metrics, count generation, PCA, DESeq2, etc (TBD)\n",
    "\n",
    "- [**Visualizations**](#Visualizations) -> Examples and ways to visualize the data and findings\n",
    "\n",
    "- [**Interpretations**](#Interpretations)-> Interpretations, take-aways to look out for\n",
    "\n",
    "\n",
    "\n",
    "For more info see \n",
    "- [RNA Translation](https://www.geeksforgeeks.org/biology/rna-translation/)\n",
    "- [Biology Homepage GfG](https://www.geeksforgeeks.org/biology/biology/)\n",
    "- [Biopython Alignment GfG](https://www.geeksforgeeks.org/python/biopython-sequence-alignment/)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0bb5537-7b3e-4df8-a645-1df2b1695812",
   "metadata": {},
   "source": [
    "## **Pre-requisites**\n",
    "\n",
    "- [python 3.10+](https://www.python.org/downloads/)\n",
    "- [pip](https://pypi.org/project/pip/)\n",
    "- [r 4.3+](https://www.r-project.org/)\n",
    "- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)\n",
    "  \n",
    "### packages\n",
    "\n",
    "- [STAR v2.7.6a](https://github.com/alexdobin/STAR)\n",
    "- [DESeq2 v1.45.0](https://github.com/thelovelab/DESeq2)\n",
    "- [dplyr v1.1.4](https://github.com/tidyverse/dplyr)\n",
    "- [Rtsne v0.17](https://github.com/jkrijthe/Rtsne)\n",
    "- [ggplot2 v3.5.1](https://github.com/tidyverse/ggplot2)\n",
    "- [ggforce v0.4.2](https://github.com/thomasp85/ggforce) **~?**\n",
    "- [pheatmap v1.0.12](https://github.com/raivokolde/pheatmap) **~?**\n",
    "- [UpSetR v1.4.0](https://github.com/hms-dbmi/UpSetR) **~?**\n",
    "- [gprofiler2 v0.2.3](https://github.com/intel/gprofiler) **~?**\n",
    "- [CIBERSORT v0.1.0](https://github.com/Moonerss/CIBERSORT) **~?**\n",
    "- [kallisto v0.44.0](https://github.com/pachterlab/kallisto) **~?**\n",
    "\n",
    "#### Note - *~?* Marked packages are optional depending on the goal.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9828e3b-dd3f-4b10-a7e7-611a3f3e5b99",
   "metadata": {},
   "source": [
    "## **Files**\n",
    "\n",
    "### FASTQ:\n",
    "The Raw reads data \n",
    "- Identifier\n",
    "- Sequence\n",
    "- Seperator\n",
    "- QC score encoded in ASCII\n",
    "- May be trimmed post QC check -> new trimmed file. Do not modify base file ever.\n",
    "\n",
    "\n",
    "### FASTQC:\n",
    "Output from QC check\n",
    "- Quality per base\n",
    "- Adapter contamination\n",
    "  \n",
    "\n",
    "### FASTA:\n",
    "- Chromosome sequences\n",
    "- Used for alignment (by STAR, for example)\n",
    "\n",
    "\n",
    "### GTF:\n",
    "- Gene locations\n",
    "- Exons\n",
    "- Transcripts\n",
    "- Used by: STAR (splice awareness) & Downstream counting\n",
    "\n",
    "\n",
    "### BAM File:\n",
    "- Each readâ€™s genomic position\n",
    "- Mapping quality\n",
    "- CIGAR strings (splicing info)\n",
    "\n",
    "\n",
    "### BAM Index:\n",
    "- Fast random access\n",
    "- Required for visualization\n",
    "\n",
    "\n",
    "### Star Log Files\n",
    "- % mapped reads\n",
    "- Unique vs multi-mapped\n",
    "- Splice junction stats\n",
    "\n",
    "\n",
    "### Count matrices:\n",
    "- csv\n",
    "- tsv\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09ee5e14-01cf-4469-97ca-5b4e9d2ebd8b",
   "metadata": {},
   "source": [
    "## Data\n",
    "\n",
    "GEO / SRA\n",
    "Look at data links in paper\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "173d705b-aa32-4b1f-9045-9e14ffcf51c7",
   "metadata": {},
   "source": [
    "## QC\n",
    "\n",
    "FASTQC MULTIQC\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "378a13c6-5c7c-471b-9f80-a815765c65c0",
   "metadata": {},
   "source": [
    "## Trimming\n",
    "\n",
    "trimming adapters/bad reads etc\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77d5e75c-a5ad-4afc-af3e-7224df02003f",
   "metadata": {},
   "source": [
    "## Alignment\n",
    "\n",
    "STAR alignment -> BAM"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00e97717-07f7-418f-9301-9220b5e3abf5",
   "metadata": {},
   "source": [
    "## Down Stream Analysis\n",
    "\n",
    "follow study for final proj, figure out the down stream analysis steps tha tmatter most/are doable and track example here\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a0e55dd-c584-43c0-ad62-5549f93857e6",
   "metadata": {},
   "source": [
    "## Visualizations\n",
    "\n",
    "visualizations of data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5231e012-f843-48b2-88b3-9aedb321e77c",
   "metadata": {},
   "source": [
    "## Interpretations\n",
    "\n",
    "interpretations of visuals/metrics/stats/data etc\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9dbefefc-f03f-49e7-92ab-0e702fe83938",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
